packages/bridge/contracts/l1/L1DaiGateway.sol

```diff
@@ -81,51 +100,99 @@ contract L1DaiGateway is L1ArbitrumExtendedGateway {
     emit Closed();
   }
 
-  function createOutboundTx(
-    address _from,
-    uint256 _tokenAmount,
-    uint256 _maxGas,
-    uint256 _gasPriceBid,
-    uint256 _maxSubmissionCost,
-    bytes memory _outboundCalldata
-  ) internal override returns (uint256) {
+  function outboundTransfer(
+    address l1Token,
+    address to,
+    uint256 amount,
+    uint256 maxGas,
+    uint256 gasPriceBid,
+    bytes calldata data
+  ) external payable returns (bytes memory) {
     // do not allow initiating new xchain messages if bridge is closed
     require(isOpen == 1, "L1DaiGateway/closed");
-
-    return
-      sendTxToL2(
-        inbox,
-        counterpartGateway,
-        _from,
-        msg.value, // we forward the L1 call value to the inbox
-        0, // l2 call value 0 by default
-        L2GasParams({
-          _maxSubmissionCost: _maxSubmissionCost,
-          _maxGas: _maxGas,
-          _gasPriceBid: _gasPriceBid
-        }),
-        _outboundCalldata
+    require(l1Token == l1Dai, "L1DaiGateway/token-not-dai");
+
+    // we use nested scope to avoid stack too deep errors
+    address from;
+    uint256 seqNum;
+    bytes memory extraData;
+    {
+      uint256 maxSubmissionCost;
+      (from, maxSubmissionCost, extraData) = parseOutboundData(data);
+      require(extraData.length == 0, "L1DaiGateway/call-hook-data-not-allowed");
+
+      TokenLike(l1Token).transferFrom(from, l1Escrow, amount);
```

Comment:
A bit of a nit pick

It might be worth declaring the `TokenLike` interface as DaiLike or something else to indicate the behaviour expected here is the behaviour of the dai token.
ie there are tokens with `transferFrom` logic that don't actually send the full `amount` so require different handling.

The gateway name also indicates this is intended for dai, so shouldn't be an issue to keep it this way